### ISSUE TITLE
all comments are rejected after update to WP 4.7.3

### ISSUE THREAD

**Issue by iamwebrocker** [pluginkollektiv/antispam-bee#144]
Possibly related to
https://wordpress.org/support/topic/all-comments-are-rejected-v-2-7-0/
and
https://github.com/pluginkollektiv/antispam-bee/issues/131

ASB Version: 2.7.0 / WP: 4.7.3 / Theme: Custom

On my blog the comments are rejected with a "please leave a comment" error message. I guess it's b/c somehow the "honeypot" textarea is recognized (by wordpress?) as a field that should have a value in it.
in my twitter timeline at least one other WP user has the same effect, while another one states that 4.7.3 and Antispambee just runs fine.
I deleted all php-caches, de- and re-activated the plugin several times, changed the wp_comment(); part to use the default settings (in my theme, I have a slightly modified markup for the form fields, including aria-required attributes) - the problem persists. if I deactivate the plugin, the comments are accepted just fine.
Any pointers what to look out for in ways how the "normal" textarea is separated from the "honeypot" textarea in terms of "which one holds the real comment"?
thx

**Comment by iamwebrocker** [comment]
I tried and applied the fix from #131 (remove the get_the_title() part), but to no avail, problem persists

**Comment by websupporter** [comment]
I @iamwebrocker,
thanks so much for your input. This is really appreciated. So basically, what this error is indicating, somehow, the method `get_secret_name_for_post()` does not return the same string in `template_redirect` (where we use it to populate the `name` attribute for the comment field, https://github.com/pluginkollektiv/antispam-bee/blob/master/antispam_bee.php#L1299) and `init` (where we use it, to read the content of the `$_POST`, https://github.com/pluginkollektiv/antispam-bee/blob/master/antispam_bee.php#L1105). At least, this is my suspicion.

I thought, the only possible way for this behavior would be the filter `the_title`, as mentioned above, but there seems to be another interference.

Basically, you have applied this commit here by hand, right? https://github.com/pluginkollektiv/antispam-bee/pull/134/commits/148e54c57c62e24d94e2aa462a704c74903e92ba Or did you just use the master-branch?

Can you reproduce this problem in a testing environment. Can you reproduce it, without using your theme? If not, is there any chance, you could send me the theme as a ZIP, so I could reproduce it locally?

Do you submit the comments via Ajax by any chance? Probably not, you wouldn't run into the `wp_die()`-message.

**Comment by iamwebrocker** [comment]
hi @websupporter thanks for time and effort and coming back!
I applied the changes regarding the get_the_title() functions by hand, just testing in an otherwise untouched 2.7.0 version of ASB.
I will try and set up an testing env; currently everything happens in production aka my live site ;-)
I ran the "check theme" test against my custom theme, and since it spat out some warnings against using  "the_title()" - which currently I'm using to generate custom html-page titles, and seemingly in a way that's not up to the current state-of-art in wp-themes terms, my guess is, that the problem lies there. I will first try to consolidate "my" title handling, i.e. moving my more or less hard coded titles to the title filter function, then check if the problem persists.
The comments are not send via Ajax btw.

**Comment by websupporter** [comment]
Hi @iamwebrocker,
thanks. My problem right now is, with applying the patch, your handling of the title shouldn't be a problem no more, since we removed `get_the_title()` to generate our secret. So even if the theme checker complains about the usage, it should not affect Antispam Bee.

Thanks for having a look into this.

**Comment by iamwebrocker** [comment]
hi @websupporter  I could give you access to my theme's github repo if you want to have a look. but be warned, it's an organic mess, constantly grown and patched since wp ~2.8 :)) and never meant to work as a general 'theme' for distribution - too many things specific to my install/blog are living in there (huge function.php, with lots of stuff that could or should be in the plugins domain) etc...
so first let me try the clean/test install, i think i can fit this in today or over the weekend latest

**Comment by websupporter** [comment]
Hi @iamwebrocker 
:heart:

this would be great. No worries, we all struggle in an organic mess.

> patched since wp ~2.8
nice :)

**Comment by iamwebrocker** [comment]
Hi @websupporter
I switched into a staging env, and swapped my theme with a default twentyfifteen, deactivated the hotfix plugin, and posted a comment -- it published right away. (http://stage.webrocker.eu/2017/03/10/css-grid-support-chrome-now-haz-it-too/comment-page-1/#comment-9152)
So there's something in my theme that's interfering - which means that this offially is no more an ASB problem.
If you're still interested, could you dm me on twitter @webrocker or send me an email (twittername and gmail com)? could give you an admin access in the stage and/or theme theme files.

**Comment by iamwebrocker** [comment]
@websupporter thank you for the stellar support! 

for the record: if your theme messes with the main query (for example, make a  ```new WP_Query($args);``` in the sidebar to get recent posts and forget to make a ```wp_reset_postdata();``` after your custom query/loop has run, the original post ID may be not available when the AntispamBee magic tries to identify the correct comment-data, thus leading to the error.
In an untempered default theme ASB has no problem. And now, in my accordingly fixed theme, it works again like a charm.
:-)

**Comment by websupporter** [comment]
Its not solved yet. Lets see if #145 works properly and lets merge it. On my tests, it is looking good :)

**Comment by websupporter** [comment]
solved via #145
